common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -12,38 +12,23 @@ class Unauthorized < StandardError; end
 
       class Forbidden < StandardError; end
 
-      #######################
-      # Constructor methods #
-      #######################
-
-      def self.for_source(source:, credentials:)
-        credential =
-          credentials.
-          select { |cred| cred["type"] == "git_source" }.
-          find { |cred| cred["host"] == source.hostname }
-
-        new(credentials: credential)
-      end
```

Comment:
I'm slightly worried that removing this might break folks using this method, can we leave it in?

Reply:
will return.